### PR TITLE
fix: apply playground improvements

### PR DESCRIPTION
**Pull request by RyukTheCoder**
# Summary

- [x] title for language setting
- [x] menu active state
- [x] reset for custom colors doesn't seem to work
- [x] disable text for max height / max width
- [x] Zoom on playground, make buttons out of screen.

# How did you test this change?

Tested this by running playground on Chrome browser and checking each of listed items.

# Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Implemented a user interface (UI) change, referencing our Figma design to ensure pixel-perfect precision.

### CODE REVIEW

File: widget/playground/src/containers/configContainer/ConfigContainer.styles.ts [hunk 1]
```diff
@@ -62,9 +61,11 @@ export const ResetButton = styled(StyledButton, {
...
-  height: '100%',
+  width: '100%',
+  flex: '1 1 0%',
+  overflow: 'auto',
   display: 'flex',
-  justifyContent: 'center',
+  justifyContent: 'safe center',
   alignItems: 'center',
   flexDirection: 'column',
 });
```
Comment (samobasquiat) [inline widget/playground/src/containers/configContainer/ConfigContainer.styles.ts:71]: these changes affected boundary guide.

**Review by samobasquiat — CHANGES_REQUESTED** [review]

**Comment by samobasquiat** [comment]
The indicator height is too large. I think it's not good. What do you think?

File: widget/playground/src/containers/configContainer/ConfigContainer.styles.ts (hunk 1 above)
Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.styles.ts:70]: fixed

**Review by samobasquiat — APPROVED** [review]

File: widget/playground/src/components/SideNavigation/SideNavigation.styles.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export const Tab = styled(Flex, {
   padding: '$15 0',
   cursor: 'pointer',
   width: '100%',
+  height: '105px',
```
Comment (yeager-eren) [inline widget/playground/src/components/SideNavigation/SideNavigation.styles.ts:35]: it's defined twice. one here and another one on `widget/playground/src/components/SideNavigation/SideNavigation.Tabs.tsx` and `TAB_HEIGHT`.

Make them using same value.

File: widget/playground/src/constants/presets.ts [hunk 3]
```diff
@@ -279,20 +279,6 @@ export const WIDGET_COLORS: { key: WidgetColorsKeys; label: string }[] = [
...
-  },
-  light: {
-    primary: undefined,
-    secondary: undefined,
-    neutral: undefined,
-    background: undefined,
-    foreground: undefined,
-    info: undefined,
-  },
+  dark: PRESETS.find((preset) => preset.id === 1)?.dark as WidgetColors,
```
Comment (yeager-eren) [inline widget/playground/src/constants/presets.ts:282]: it can be `undefined` (based on `find(...)?`) but you have added a type assertion which eliminated the `undefined`.

Comment (yeager-eren) [reply widget/playground/src/constants/presets.ts:282]: Please also don't use magic values which means we prefer `const DEFAULT_PRESET_ID = 1;` over using a magic number directly.

File: widget/playground/src/containers/configContainer/ConfigContainer.styles.ts [hunk 4]
```diff
@@ -62,9 +61,10 @@ export const ResetButton = styled(StyledButton, {
...
 
 export const Content = styled('div', {
-  height: '100%',
+  flex: '1 1 0%',
+  overflow: 'auto',
   display: 'flex',
-  justifyContent: 'center',
+  justifyContent: 'safe center',
```
Comment (yeager-eren) [inline widget/playground/src/containers/configContainer/ConfigContainer.styles.ts:67]: Is this a thing in CSS?

File: widget/ui/src/components/Typography/Typography.styles.tsx [hunk 5]
```diff
@@ -33,6 +33,11 @@ export const TypographyContainer = styled('span', {
         whiteSpace: 'nowrap',
       },
     },
+    selectNone: {
```
Comment (yeager-eren) [inline widget/ui/src/components/Typography/Typography.styles.tsx:36]: Please don't every css property a prop. and also make it boolean will cause to add `selectAll`, `selectText` and other possible values in future.

Comment (yeager-eren) [reply widget/ui/src/components/Typography/Typography.styles.tsx:36]: You can cover you use case in playground by doing something like this.

```
const NotSelectableTypograph = styled(Typograph, { ... })
```

**Review by yeager-eren — CHANGES_REQUESTED** [review]

**Comment by RyukTheCoder** [comment]
> The indicator height is too large. I think it's not good. What do you think?

File: widget/playground/src/components/SideNavigation/SideNavigation.styles.ts (hunk 2 above)
Comment (RyukTheCoder) [reply widget/playground/src/components/SideNavigation/SideNavigation.styles.ts:35]: Thanks, Fixed.

File: widget/playground/src/constants/presets.ts (hunk 3 above)
Comment (RyukTheCoder) [reply widget/playground/src/constants/presets.ts:282]: Thanks, I changed my code to use the "DEFAULT_PRESET_ID" that you suggested. About the other comment, the preset with id equal to 1 is hard coded so it is not possible for the result of the find method to be 'undefined' so I don't get the problem here.

File: widget/playground/src/containers/configContainer/ConfigContainer.styles.ts (hunk 4 above)
Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.styles.ts:67]: Yes, it is commonly used to address issues with a scrollable flex container. Let's consider a scenario where you have a div element with 'display' set to 'flex', 'justifyContent' set to 'center', and 'overflow' set to 'auto'. If this div element contains a child element with a height greater than the parent, the div element becomes scrollable. However the upper section of the child element may be hidden. To overcome this problem, you can utilize the 'safe center' property.

File: widget/ui/src/components/Typography/Typography.styles.tsx (hunk 5 above)
Comment (RyukTheCoder) [reply widget/ui/src/components/Typography/Typography.styles.tsx:36]: Added "NotSelectableTypograph" as a component to UI repository.

File: widget/playground/src/containers/configContainer/ConfigContainer.styles.ts (hunk 4 above)
Comment (yeager-eren) [reply widget/playground/src/containers/configContainer/ConfigContainer.styles.ts:67]: Thanks. I didn't know it exists.
It seems it doesn't has satisfying browser support yet, but I tested it in all major browsers and everything was fine.

File: widget/playground/src/utils/colors.ts [hunk 6]
```diff
@@ -10,15 +12,58 @@ export const getMainColor = (
   const isDarkTab = tab === 'dark';
   const isLightTab = tab === 'light';
   const isAutoTab = tab === 'auto';
+
   if (!!theme?.colors) {
-    if (isAutoTab && !theme.singleTheme) {
-      return (theme?.colors[mode || 'light'] || {})[key];
+    if (isAutoTab) {
```
Comment (yeager-eren) [inline widget/playground/src/utils/colors.ts:17]: As we talked in tg, you can revert this change and only disable `Custom Color` when the selected preset isn't in the current tab.

Comment (RyukTheCoder) [reply widget/playground/src/utils/colors.ts]: Done

File: widget/playground/src/containers/configContainer/ConfigContainer.tsx [hunk 7]
```diff
@@ -67,29 +68,37 @@ export function ConfigContainer(props: PropsWithChildren) {
...
-                <Typography variant="label" size="medium" color="neutral600">
-                  Max Width: 390 px
-                </Typography>
-              </BoundarySize>
-            )}
-            <BoundaryGuide
-              style={{ borderWidth: showBoundaryGuide ? '1px' : 0 }}>
-              {props.children}
-            </BoundaryGuide>
+            <div>
```
Comment (yeager-eren) [inline widget/playground/src/containers/configContainer/ConfigContainer.tsx:71]: Why this `div` needed?

Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.tsx:71]: To address the issue depicted in the image, a container needs to be created to properly align the texts related to the boundary guide. The container should have a width that fits the dimensions of the boundary guide to ensure correct placement.

image

Comment (yeager-eren) [reply widget/playground/src/containers/configContainer/ConfigContainer.tsx:71]: It's kinda weird, but if it's need, please put a comment on top of it and explain the reason because it can be removed in future easily since removing it will cause problem on a specific situation.

Comment (RyukTheCoder) [reply widget/playground/src/containers/configContainer/ConfigContainer.tsx:71]: Done

**Review by yeager-eren — APPROVED** [review]
Thanks. I'll merge it.
